lib/rubocop/cop/solidus/class_eval_decorator.rb

```diff
@@ -38,6 +18,7 @@ class ClassEvalDecorator < Base
 
         def on_send(node)
           return unless on_class_eval?(node)
+          return unless minimum_supported_solidus_version?(3.0)
```

Comment:
This call should stop execution of the cop and return without adding any offenses.

Only implementation that is close to working as other implementations require overriding rubocop classes which I haven't been particularly successful in getting to work correctly.